File: AdminBackend/src/main/java/Backend/AdminBackend/model/Sertifikat.java
package Backend.AdminBackend.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.security.PublicKey;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Sertifikat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String alias;

    @Column
    private String namena;

    @Column
    private PublicKey publicKey;

    @Column
    private Date startDate;

    @Column
    private Date endDate;

    @Column
    private  String subjectEmail;

    @Column
    private String organizacionaJedinica;

    @Column
    private String nazivOrganizacije;

    @Column
    private String skraceniNazivZemlje;

    @OneToOne(targetEntity = ZaKorisnika.class, fetch = FetchType.EAGER)
    @JoinColumn(nullable = true)
    private ZaKorisnika zaKorisnika;

    @OneToOne(targetEntity = ZaUredjaj.class, fetch = FetchType.EAGER)
    @JoinColumn(nullable = true)
    private ZaUredjaj zaUredjaj;

    @OneToOne(targetEntity = ZaMojaKucaAplikacija.class, fetch = FetchType.EAGER)
    @JoinColumn(nullable = true)
    private ZaMojaKucaAplikacija zaMojaKucaAplikacija;

    @ManyToOne(cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    @JoinColumn
    private Korisnik musterija;

    @ManyToOne(cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    @JoinColumn
    private Korisnik admin;

    @OneToOne(cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    @JoinColumn
    private Ekstenzije ekstenzije;

    public Sertifikat(String alias, String namena) {
        this.alias = alias;
        this.namena = namena;
    }
}

File: AdminFrontend/src/app/MODEL/Sertifikat.ts
import { Ekstenzije } from "./Ekstenzije";
import { Korisnik } from "./Korisnik";
import { ZaKorisnika } from "./ZaKorisnika";
import { ZaMojaKucaAplikacija } from "./ZaMojaKucaAplikacija";
import { ZaUredjaj } from "./ZaUredjaj";

export class Sertifikat {
    alias: string
    namena: string
    startDate: Date
    endDate: Date
    subjectEmail: string

    organizacionaJedinica: string
    nazivOrganizacije: string
    skraceniNazivZemlje: string

    zaKorisnika: ZaKorisnika
    zaUredjaj: ZaUredjaj
    zaMojaKucaAplikacija: ZaMojaKucaAplikacija
    musterija: Korisnik
    admin: Korisnik
    ekstenzije: Ekstenzije
    constructor(
        alias: string,
        namena: string,
        startDate: Date,
        endDate: Date,
        subjectEmail: string,
        organizacionaJedinica: string,
        nazivOrganizacije: string,
        skraceniNazivZemlje: string,
        zaKorisnika: ZaKorisnika,
        zaUredjaj: ZaUredjaj,
        zaMojaKucaAplikacija: ZaMojaKucaAplikacija,
        musterija: Korisnik,
        admin: Korisnik,
        ekstenzije: Ekstenzije
    ) {
        this.alias = alias
        this.namena = namena
        this.startDate = startDate
        this.endDate = endDate
        this.subjectEmail = subjectEmail
        this.organizacionaJedinica = organizacionaJedinica
        this.nazivOrganizacije = nazivOrganizacije
        this.skraceniNazivZemlje = skraceniNazivZemlje
        this.zaKorisnika = zaKorisnika
        this.zaUredjaj = zaUredjaj
        this.zaMojaKucaAplikacija = zaMojaKucaAplikacija
        this.musterija = musterija
        this.admin = admin
        this.ekstenzije = ekstenzije
    }

}